src/Models/Tag.php

```diff
@@ -15,18 +15,20 @@
 
 class Tag extends BaseModel implements Sortable
 {
-    use SortableTrait, HasSlug, HasCreator, HasUpdater, HasPackageFactory;
+    use SortableTrait, HasCreator, HasUpdater, HasPackageFactory;
 
     protected $guarded = ['id'];
 
-    protected static function boot()
+    public static function boot()
     {
         parent::boot();
 
-        static::saving(function ($tag) {
-            if (empty($tag->slug)) {
-                $tag->slug = $tag->generateSlug();
-            }
+        static::saving(static function ($tag) {
+            $sanitizedName = Str::keepAlphanumericCharacters($tag->name);
+            $splitUpName = Str::splitAtCapitalLetters($sanitizedName);
+
+            $tag->slug = Str::slug($splitUpName);
+            $tag->name = '#' . Str::studly($splitUpName);
         });
```

Comment:
This is a clean way of sanitizing the tag name and generating the slug. I like it!